spring-boot-project/spring-boot-docs/src/main/asciidoc/howto.adoc

```diff
@@ -2108,26 +2116,69 @@ See {spring-batch-api}/core/configuration/annotation/EnableBatchProcessing.html[
 
 For more about Spring Batch, see the {spring-batch}[Spring Batch project page].
 
-
-
-[[howto-execute-spring-batch-jobs-on-startup]]
-=== Execute Spring Batch Jobs on Startup
+[[howto-run-spring-batch-jobs-on-startup]]
+=== Running Spring Batch Jobs on Startup
 Spring Batch auto-configuration is enabled by adding `@EnableBatchProcessing` (from Spring Batch) somewhere in your context.
 
 By default, it executes *all* `Jobs` in the application context on startup (see {spring-boot-autoconfigure-module-code}/batch/JobLauncherCommandLineRunner.java[JobLauncherCommandLineRunner] for details).
 You can narrow down to a specific job or jobs by specifying `spring.batch.job.names` (which takes a comma-separated list of job name patterns).
 
-[TIP]
-.Specifying job parameters on the command line
-====
-Unlike command line option arguments that <<spring-boot-features.adoc#boot-features-external-config-command-line-args,set properties in the `Environment`>> (i.e. by starting with `--`, such as `--my-property=value`), job parameters have to be specified on the command line without dashes (e.g. `jobParam=value`).
-====
-
-If the application context includes a `JobRegistry`, the jobs in `spring.batch.job.names` are looked up in the registry instead of being autowired from the context.
-This is a common pattern with more complex systems, where multiple jobs are defined in child contexts and registered centrally.
-
 See {spring-boot-autoconfigure-module-code}/batch/BatchAutoConfiguration.java[BatchAutoConfiguration] and https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/configuration/annotation/EnableBatchProcessing.java[@EnableBatchProcessing] for more details.
 
+[[howto-spring-batch-running-command-line]]
+=== Running from the Command Line
+
+Running Spring Batch with Spring Boot from the command line differs from running Spring
+Batch by itself from the command line. Spring Boot uses a class called
+`JobLaunchingCommandLineRunner`. Spring Batch uses a class called `CommandLineJobRunner`.
+They work a bit differently. However, for the most part, the differences do not cause
+trouble. However, they parse command line arguments differently, which can cause trouble,
+as described in the next section.
+// TODO What are the other differences between the two?
```

Comment:
Is that something we're supposed to handle?

Reply:
The other difference is that the `CommandLineJobRunner` from batch can accepts options like `-next`, `-stop`, `-restart`, etc while the one from boot does not.

Reply:
I left out that detail, since I didn't think it would help a user of Spring Boot.